demo/site/src/app/[domain]/[language]/news/api/route.ts

```diff
@@ -0,0 +1,24 @@
+import { fetchNewsList } from "@src/news/NewsPage.loader";
+import { type NextRequest, NextResponse } from "next/server";
+import { z } from "zod";
+
+import { PageParams } from "../page";
+
+export async function GET(request: NextRequest, { params }: { params: PageParams }) {
+    const result = z
+        .object({
+            offset: z.coerce.number().min(0),
+            limit: z.coerce.number().min(1).max(100),
+        })
```

Comment:
we might specify the object its shape outside of the handler

(might be nit picking or personal preference or "class-validator" style)

Reply:
The validation schema isn't generated on each API call, but this is probably premature optimization.